src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,144 @@
+
+                } else if (firstWord.equals("unmark")) {
+                    //Second word should be an integer dictating which Task to unmark.
+                    String secondWord = userInput.split(" ", 2)[1];
+                    int taskNumber = Integer.parseInt(secondWord);
+                    Task task = tasks.getTask(taskNumber);
+                    storage.changeTaskStatus(task.getStorageLine());
+                    tasks.unmark(taskNumber);
+                    ui.respond("I have marked this task as undone! \n" + task.provideDetails());
+
+                } else if (firstWord.equals("todo")) {
+                    //Rest of message describes the Task.
+                    String body = userInput.split(" ", 2)[1];
+                    ToDo task = new ToDo("todo", body, false);
+                    tasks.addTask(task);
+                    storage.addTask(task.getStorageLine());
+
+                    ui.respond("I have added this new task:\n" + task.provideDetails()
+                            + "\nYou now currently have "
+                            + tasks.getTaskCount() + " tasks.");
+
+                } else if (firstWord.equals("deadline")) {
+                    //Rest of message describes the Task.
+                    String body = userInput.split(" ", 2)[1];
+                    DeadLine newTask = new DeadLine("deadline", body, false);
+                    tasks.addTask(newTask);
+                    storage.addTask(newTask.getStorageLine());
+                    ui.respond("I have added this new task:\n" + newTask.provideDetails()
+                            + "\nYou now currently have "
+                            + tasks.getTaskCount() + " tasks.");
+
+                } else if (firstWord.equals("event")) {
+                    //Rest of message describes the Task.
+                    String body = userInput.split(" ", 2)[1];
+                    Event newTask = new Event("event", body, false);
+                    tasks.addTask(newTask);
+                    storage.addTask(newTask.getStorageLine());
+
+                    ui.respond("I have added this new task:\n" + newTask.provideDetails()
+                            + "\nYou now currently have "
+                            + tasks.getTaskCount() + " tasks.");
+
+
+                } else if (firstWord.equals("delete")) {
+                    //second word should be an integer
+                    String secondWord = userInput.split(" ", 2)[1];
+                    int taskNumber = Integer.parseInt(secondWord);
+                    Task task = tasks.getTask(taskNumber);
+                    tasks.deleteTask(taskNumber);
+                    storage.deleteTask(task.getStorageLine());
+
+                    ui.respond("We have removed this task: " + task.provideDetails() + "\nYou now have "
+                            + tasks.getTaskCount() + " tasks remaining");
+
+                } else {
+                    ui.showCommandError();
+                }
```

Comment:
Maybe consider refactoring the code here to be stored into the parser class?

Reply:
Thanks! I missed out the Parser class while completing this increment. @thennant